storey/utils.py

```diff
@@ -42,6 +43,14 @@ def get_one_unit_of_duration(string_time):
     return multiplier
 
 
+def get_window_start_time_by_time(timestamp, window_millis):
+    midnight = datetime.utcfromtimestamp(int(timestamp/1000)).replace(hour=0, minute=0, second=0)
+    window_seconds = int(window_millis / 1000)
+    timestamp_seconds = int(timestamp / 1000)
+    return max(int(timestamp_seconds / window_seconds) * window_seconds,
+               midnight.replace(tzinfo=timezone.utc).timestamp()) * 1000
```

Comment:
Might as well replace the time zone together with the hour/minute/second, no?

Reply:
Done